share/p2p/metrics.go

```diff
@@ -0,0 +1,66 @@
+package p2p
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric/global"
+	"go.opentelemetry.io/otel/metric/instrument"
+	"go.opentelemetry.io/otel/metric/instrument/syncint64"
+	"go.opentelemetry.io/otel/metric/unit"
+)
+
+var meter = global.MeterProvider().Meter("shrex/eds")
+
+type status string
+
+const (
+	StatusInternalErr status = "InternalError"
+	StatusNotFound    status = "NotFound"
+	StatusSuccess     status = "Success"
+	StatusRateLimited status = "RateLimited"
+)
+
+type Metrics struct {
+	totalRequestCounter syncint64.Counter
+}
+
+// ObserveRequests increments the total number of requests sent with the given status as an attribute.
+func (m *Metrics) ObserveRequests(ctx context.Context, count int64, status status) {
+	if m == nil {
+		return
+	}
+
+	m.totalRequestCounter.Add(ctx, count, attribute.String("status", string(status)))
+}
+
+func InitClientMetrics(protocol string) (*Metrics, error) {
+	totalRequestCounter, err := meter.SyncInt64().Counter(
+		fmt.Sprintf("total_shrex_%s_requests", protocol),
```

Comment:
Pls reorder name so it is easier to navigate in metrics explorer:
`module_submodule_metricname`
So in this case it would be:
`shrex_client_total...`

Pls update other metrics names accordingly.